### ISSUE TITLE
Update boost headers

### ISSUE THREAD

**Issue by ggardet** [ARM-software/armnn#274]
With latest boost, you currently need to apply the following patch to build armNN tests:
```
--- armnn-19.02.orig/src/armnn/test/TensorHelpers.hpp	2019-03-26 16:08:56.403605790 +0100
+++ armnn-19.02/src/armnn/test/TensorHelpers.hpp	2019-03-26 16:09:58.448265384 +0100
@@ -13,6 +13,7 @@
 #include <boost/assert.hpp>
 #include <boost/test/tools/floating_point_comparison.hpp>
 #include <boost/random/uniform_real_distribution.hpp>
+#define BOOST_ALLOW_DEPRECATED_HEADERS
 #include <boost/random/mersenne_twister.hpp>
 #include <boost/numeric/conversion/cast.hpp>
```
You probably need to update some headers to match the new layout.

**Comment by MatthewARM** [comment]
Which headers are deprecated? Can we somehow remove our usage of them entirely?

**Comment by ggardet** [comment]
According to the log message:
```
[  278s] [ 79%] Building CXX object CMakeFiles/UnitTests.dir/src/armnn/test/ProfilerTests.cpp.o
[  278s] /usr/bin/c++  -DARMCOMPUTECL_ENABLED -DARMNNREF_ENABLED -DARMNN_CAFFE_PARSER -DARMNN_SERIALIZER -DARMNN_SERIALIZER_SCHEMA_PATH=\"/home/abuild/rpmbuild/BUILD/armnn-19.08/src/armnnSerializer/ArmnnSchema.fbs\" -DARMNN_TF_LITE_PARSER -DARMNN_TF_LITE_SCHEMA_PATH=\"/usr/include/tensorflow/lite/schema/schema.fbs\" -DARMNN_TF_PARSER -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_ATOMIC_DYN_LINK -DBOOST_CHRONO_DYN_LINK -DBOOST_DATE_TIME_DYN_LINK -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_LOG_DYN_LINK -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_REGEX_DYN_LINK -DBOOST_SYSTEM_DYN_LINK -DBOOST_THREAD_DYN_LINK -DBOOST_UNIT_TEST_FRAMEWORK_DYN_LINK -DPROTOBUF_USE_DLLS -DWITH_VALGRIND=1 -I/home/abuild/rpmbuild/BUILD/armnn-19.08/include -I/home/abuild/rpmbuild/BUILD/armnn-19.08/src/armnn -I/home/abuild/rpmbuild/BUILD/armnn-19.08/src/armnnUtils -I/home/abuild/rpmbuild/BUILD/armnn-19.08/src/backends -isystem /usr/lib/python3.7/site-packages/tensorflow/include -isystem /usr/include/half -isystem /home/abuild/rpmbuild/BUILD/armnn-19.08/third-party -isystem /usr/include/tensorflow/lite/schema -isystem /home/abuild/rpmbuild/BUILD/armnn-19.08/build/src/armnnSerializer -isystem /home/abuild/rpmbuild/BUILD/armnn-19.08/build/src/armnnQuantizer  -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type  -pthread -std=c++14 -Wall -Werror -Wold-style-cast -Wno-missing-braces -Wconversion  -O2 -g  -fPIE   -o CMakeFiles/UnitTests.dir/src/armnn/test/ProfilerTests.cpp.o -c /home/abuild/rpmbuild/BUILD/armnn-19.08/src/armnn/test/ProfilerTests.cpp
[  278s] In file included from /usr/include/boost/config/header_deprecated.hpp:18,
[  278s]                  from /usr/include/boost/test/output_test_stream.hpp:13,
[  278s]                  from /home/abuild/rpmbuild/BUILD/armnn-19.08/src/armnn/test/ProfilerTests.cpp:7:
[  278s] /usr/include/boost/test/output_test_stream.hpp:14:1: note: #pragma message: This header is deprecated. Use This header is deprecated. Please use <boost/test/tools/output_test_stream.hpp> instead. instead.
[  278s]    14 | BOOST_HEADER_DEPRECATED( "This header is deprecated. Please use <boost/test/tools/output_test_stream.hpp> instead." )
[  278s]       | ^~~~~~~~~~~~~~~~~~~~~~~
```
`boost/test/output_test_stream.hpp` should be replaced by `boost/test/tools/output_test_stream.hpp`

**Comment by MatthewARM** [comment]
Thanks, looks like that shouldn't be too hard to fix

**Comment by MatthewARM** [comment]
Looks like the new location was available since Boost 1.59, so I've just changed it unconditionally. This should be all fixed now.
